File: app/src/main/java/tech/inscripts/ins_armman/mMitra/incompleteForm/IncompleteFormInteractor.java
package tech.inscripts.ins_armman.mMitra.incompleteForm;

import android.content.Context;
import android.database.Cursor;

import com.inscripts.ins_armman.npdsf.database.DBHelper;

/**
 * @author Aniket & Vivek  Created on 4/9/2018
 */

public class IncompleteFormInteractor implements IncompleteInteractor {

    private Context mContext;
    private DBHelper dbHelper;

    IncompleteFormInteractor(Context context) {
        this.mContext = context;
        dbHelper = new DBHelper(context);
    }

    @Override
    public Cursor fetchListIncompleteForm() {
        return dbHelper.getIncompleteFormListList();
    }

    @Override
    public Cursor fetchUniqueIdChildId(String unique) {
        return dbHelper.getChildIdFromMotherId(unique);
    }

    @Override
    public Cursor checkChildFilledForm(String unique) {
        return dbHelper.getuniqueIdFormId(unique);
    }
}

File: app/src/main/java/tech/inscripts/ins_armman/mMitra/incompleteForms/IncompleteFormInteractor.kt
package tech.inscripts.ins_armman.mMitra.incompleteForms

import android.content.Context
import android.database.Cursor
import tech.inscripts.ins_armman.mMitra.data.database.DBHelper

class IncompleteFormInteractor : IIncompleteInteractor {
    override fun getLastCompleteFilledForm(uniqueId: String): Cursor {
        return dbHelper?.getLastCompleteFilledFormId(uniqueId)!!
    }

    var context : Context?=null
    var dbHelper : DBHelper?=null

    constructor(context: Context?) {
        this.context = context
        dbHelper = DBHelper(context)
    }


    override fun getIncompleteFormList(): Cursor {
        return dbHelper?.getIncompleteFormListList()!!
    }
}